sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -246,6 +246,12 @@ def send_batch(self, event_data_batch, **kwargs):
         """
         partition_id = kwargs.get("partition_id")
         partition_key = kwargs.get("partition_key")
+
+        if partition_key and not isinstance(partition_key, str):
+            _LOGGER.info("WARNING: Please DO NOT pass a partition_key of non-string type to the send_batch method."
```

Comment:
why not `_looger.warning`?

Reply:
this applies everywhere - is there a specific reason why you restricted it to info than a warning

Reply:
according to our guideline warning is used to indicate error and error should be raised:
> DO use the WARNING logging level when a function fails to perform its intended task. The function should also raise an exception.

while in special case passing partition_key of non-string type doesn't cause any error, the service simply ignore, so making it a logging info.

Reply:
Yeah I think `info` is probably appropriate here - we can bump it up a level later if we get feedback, or messaging from the service team that this should be treated more like an error. Though currently the service is not treating this like an error.
Do we want to tell people _not_ to do this? Or be informative, e.g. "By settings a non-string value, this partition key will be ignored by the service"
(Just asking, not suggesting :) )

Reply:
based on:
- what Serkant mentioned in the issue #12353 :
> Service doesn't do any validation on the delivered type and assumes sender library already conducted sanitary checks. This applies to every entry in the annotations map as far as I know.
- the customer described having trouble with Azure Function receiving messages which uses SDKs from a different language trying to parse the value:
```Unable to cast object of type 'System.Int32' to type 'System.String```

I think users had better not to set the partition_key of non-string type.

However, the wording could be improved in a "softer" way, I'm thinking of this way to be less imperative and more descriptive.

`Setting a partition_key of non-string type is discouraged in which case the partition_key will be ignored by the Event Hub service ignores and events will be assigned to all partitions using round-robin. Furthermore, SDKs from other languages might fail to parse the non-string value if they only expect partition_key to be string type`

How does it look to you?

Reply:
Looks good! Thanks!